Pod/Classes/CCNMigrationQueue.m

```diff
@@ -44,27 +45,65 @@ - (instancetype)init
 	return nil;
 }
 
+- (BOOL)checkForExecutions:(CIRDatabase *)database
+{
+	[self calculateExecutionsCountForDatabase:database];
+
+	return _executionsCount > 0;
+}
+
+- (NSUInteger)executionsCountForDatabase:(CIRDatabase *)database
+{
+	[self calculateExecutionsCountForDatabase:database];
+
+	return _executionsCount;
+}
+
+- (void)calculateExecutionsCountForDatabase:(CIRDatabase *)database
+{
+	NSUInteger count = 0;
+
+	[database executeUpdate:@"CREATE TABLE IF NOT EXISTS 'schema_migrations' ('version' INTEGER PRIMARY KEY)"];
+
+	CIRResultSet *resultSet = [database executeQuery:[NSString stringWithFormat:@"SELECT COUNT(version) FROM schema_migrations"]];
+	if ([resultSet next])
+		count = (NSUInteger) [resultSet intAtIndex:0];
+
+	_executionsCount = _migrations.count - count;
+}
+
 - (void)execute:(CIRDatabase *)database
 {
+	[self execute:database progress:nil];
+}
+
+- (void)execute:(CIRDatabase *)database progress:(void (^)(CCNAbstractMigration *, int, int))progress;
```

Comment:
What will happen if I call the `execute:` without call the `calculateExecutions...`? The table will never be created, right?